src/components/Details/VmDetails/VmDetails.js

```diff
@@ -123,10 +129,17 @@ export class VmDetails extends React.Component {
   isFormValid = () => Object.keys(this.state.form).every(key => this.state.form[key].valid);
 
   componentDidUpdate() {
-    const { launcherPod, importerPods, migration, vm } = this.props;
+    const { launcherPod, importerPods, migration, k8sGet, vm } = this.props;
     if (this.state.editing && !this.isVmOff(vm, launcherPod, importerPods, migration)) {
       this.setEditing(false);
     }
+
+    // Check template, if template is missing, update view.
+    if (getVmTemplate(vm) != null && this.state.templateError === PENDING_STR) {
+      retrieveVmTemplate(k8sGet, vm)
```

Comment:
retrieveVmTemplate is already called for the flavor. Please call this method only once.

Reply:
:+1: